src/bentoml/_internal/server/metrics/prometheus.py

```diff
@@ -108,59 +135,343 @@ def generate_latest(self):
             return self.prometheus_client.generate_latest()
 
     def text_string_to_metric_families(self) -> t.Generator[Metric, None, None]:
```

Comment:
Should this be exposed to users?

Reply:
yes. Because if they want to process metrics from their custom collector, they should use this function instead of the one from `prometheus_client`.